lib/slots.ts

```diff
@@ -17,11 +17,11 @@ export type GetSlots = {
   minimumBookingNotice: number;
 };
 
-const getMinuteOffset = (date: Dayjs, step: number) => {
+const getMinuteOffset = (date: Dayjs, frequency: number) => {
   // Diffs the current time with the given date and iff same day; (handled by 1440) - return difference; otherwise 0
-  const minuteOffset = Math.min(date.diff(dayjs.utc().startOf("day"), "minute"), 1440) % 1440;
+  const minuteOffset = Math.min(date.diff(dayjs().utc(), "minute"), 1440) % 1440;
   // round down to nearest step
```

Comment:
Thanks @emrysal  for pointing out to check the tests.

The main issue here was using the date with utc together with `.startOf`
Here is my observation. I'm at UTC +8 (Asia/Kuala_Lumpur). When you use the `date.utc().start("day")` date would look like `2022-01-13 00:08:00 +8`, that's basically doubles the minutesOffset value from 480 (8 hours *60) to 960. This affects the time slots lists for the following day. Hence the above change.